robot/web/stream/server.go

```diff
@@ -2,65 +2,52 @@ package webstream
 
 import (
 	"context"
-	"errors"
 	"fmt"
 	"sync"
 
 	"github.com/pion/webrtc/v3"
+	"github.com/pkg/errors"
 	"go.uber.org/multierr"
 	streampb "go.viam.com/api/stream/v1"
 	"go.viam.com/utils"
 	"go.viam.com/utils/rpc"
 
 	"go.viam.com/rdk/gostream"
+	"go.viam.com/rdk/logging"
+	"go.viam.com/rdk/robot"
+	"go.viam.com/rdk/robot/web/stream/state"
+	rutils "go.viam.com/rdk/utils"
 )
 
 type peerState struct {
-	stream  *streamState
-	senders []*webrtc.RTPSender
-}
-
-type streamState struct {
-	mu          sync.Mutex
-	stream      gostream.Stream
-	activePeers int
-}
-
-func (ss *streamState) Start() {
-	ss.mu.Lock()
-	defer ss.mu.Unlock()
-	ss.activePeers++
-	if ss.activePeers == 1 {
-		ss.stream.Start()
-	}
-}
-
-func (ss *streamState) Stop() {
-	ss.mu.Lock()
-	defer ss.mu.Unlock()
-	ss.activePeers--
-	if ss.activePeers <= 0 {
-		ss.activePeers = 0
-		ss.stream.Stop()
-	}
```

Comment:
This code has been moved into https://github.com/viamrobotics/rdk/pull/3792/files#diff-711be8ef6ac79c9940f9d8b2cd13c350370bd347b985236bd4cd5c9253659c10